### PR TITLE
HHH-13882 Fix bug in EntityGraphQueryHint

### PR DESCRIPTION
**Pull request by NathanQingyangXu**
https://hibernate.atlassian.net/browse/HHH-13882

The bug manifests itself with the following preconditions:

1. entity graph is used on collection
2. LockMode is specified for query

The root cause is we set 'classAlias' of LHS to RHS in EntityGraphQueryHint. Luckily EntityGraphQueryHint will be refactored in v6 so this PR only applies for v5.

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/engine/query/spi/EntityGraphQueryHint.java [hunk 1]
```diff
@@ -143,7 +141,7 @@ else if ( propertyType.isCollectionType() ) {
 
 						final FromElementFactory fromElementFactory = new FromElementFactory(
 								fromClause, origin,
-								attributeName, classAlias, columns, false
+								attributeName, null, columns, false
```
Comment (NathanQingyangXu) [inline hibernate-core/src/main/java/org/hibernate/engine/query/spi/EntityGraphQueryHint.java]: Obviously, we should not use `classAlias` here for it is the joining lhs and using it only ends up with screwing up internal `formElementByClassAlias` map in `FromClause` by overwriting lhs by a collection from element without persister, thus the NPE bug when `QueryLoader#applyLocks()` is invoked.

**Comment by wgtmpeters** [comment]
Will this ever be merged or is there a workaround?

**Comment by NathanQingyangXu** [comment]
> Will this ever be merged or is there a workaround?

This is a simple and effective fix. @beikov do you have bandwidth to take a look?

**Comment by beikov** [comment]
Could you please rebase this PR @NathanQingyangXu ?

**Pull request closed without merging**